src/main/java/org/opensearch/geospatial/search/aggregations/bucket/geogrid/GeoHexHelper.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.search.aggregations.bucket.geogrid;
+
+import static org.opensearch.geospatial.h3.H3.MIN_H3_RES;
+import static org.opensearch.geospatial.h3.H3.h3IsValid;
+import static org.opensearch.geospatial.h3.H3.h3ToLatLng;
+import static org.opensearch.geospatial.h3.H3.stringToH3;
+
+import java.util.Locale;
+
+import lombok.NonNull;
+
+import org.opensearch.common.geo.GeoPoint;
+import org.opensearch.geospatial.h3.H3;
+
+public class GeoHexHelper {
+
+    /**
+     * Checks whether given precision is within H3 Precision range
+     * @param precision H3 index precision
+     */
+    public static void checkPrecisionRange(int precision) {
+        if ((precision < MIN_H3_RES) || (precision > H3.MAX_H3_RES)) {
+            throw new IllegalArgumentException(
+                String.format(
+                    Locale.getDefault(),
+                    "Invalid precision of %d . Must be between %d and %d.",
+                    precision,
+                    MIN_H3_RES,
+                    H3.MAX_H3_RES
+                )
+            );
+        }
+    }
+
+    /**
+     * Converts from <code>long</code> representation of an index to {@link GeoPoint} representation.
+     * @param h3CellID H3 Cell Id
+     * @throws IllegalArgumentException if invalid h3CellID is provided
+     */
+    public static GeoPoint h3ToGeoPoint(long h3CellID) {
+        if (!h3IsValid(h3CellID)) {
```

Comment:
Avoid negation and be explicit. 
```suggestion
        if (h3IsValid(h3CellID) == false) {
```

> [nknize](https://github.com/nknize) [2 days ago](https://github.com/opensearch-project/OpenSearch/pull/4597#discussion_r991379077)
> nit: as a matter of convention we try to be as explicit as possible and avoid negation logic. Try to refactor away negations if you see this other places in the code base

Reply:
Ack